pkg/cmd/pulumi/deployment_settings_config.go

```diff
@@ -361,58 +432,65 @@ func configureGit(ctx context.Context, displayOpts *display.Options, be backend.
 	}
 	sd.DeploymentSettings.SourceContext.Git.RepoDir = repoDir
 
-	repo, err := git.PlainOpen(repoRoot)
+	var branchName string
+	if sd.DeploymentSettings.SourceContext.Git.Branch != "" {
+		branchName = sd.DeploymentSettings.SourceContext.Git.Branch
+	} else {
+		branchName = rl.GetBranchName()
+	}
+
+	if branchName != "" {
+		branchName, err = cmdutil.ReadConsoleWithDefault("Enter the branch name", branchName)
+	} else {
+		branchName, err = cmdutil.ReadConsole("Enter the branch name")
+	}
 	if err != nil {
 		return err
 	}
+	sd.DeploymentSettings.SourceContext.Git.Branch = branchName
 
-	h, err := repo.Head()
+	remoteURL, err := rl.RemoteURL()
 	if err != nil {
 		return err
 	}
 
-	var branch string
-	if sd.DeploymentSettings.SourceContext.Git.Branch != "" {
-		branch, err = cmdutil.ReadConsoleWithDefault("Enter the branch name", sd.DeploymentSettings.SourceContext.Git.Branch)
-	} else if h.Name().IsBranch() {
-		branch, err = cmdutil.ReadConsoleWithDefault("Enter the branch name", h.Name().String())
+	if remoteURL != "" {
+		remoteURL, err = cmdutil.ReadConsoleWithDefault("Enter the repository URL", remoteURL)
 	} else {
-		branch, err = cmdutil.ReadConsole("Enter the branch name")
+		remoteURL, err = cmdutil.ReadConsole("Enter the repository URL")
 	}
 	if err != nil {
 		return err
 	}
-	sd.DeploymentSettings.SourceContext.Git.Branch = branch
 
-	remoteURL, err := gitutil.GetGitRemoteURL(repo, "origin")
-	if err != nil {
-		return err
-	}
+	vcsInfo, err := gitutil.TryGetVCSInfo(remoteURL)
 
-	if vcsInfo, err := gitutil.TryGetVCSInfo(remoteURL); err == nil {
-		useGiHub := vcsInfo.Kind == gitutil.GitHubHostName
+	var useGiHub bool
```

Comment:
```suggestion
	var useGitHub bool
```

(affects everything below as well of course)

Reply:
done